tests/vmemcache_fragmentation.c

```diff
@@ -0,0 +1,280 @@
+		.segment_size = 16,
+		.val_max = 0,
+		.dir = "/tmp",
+		.seconds = 0,
+		.minutes = 0,
+		.hours = 0
+	};
+	size_t val_max_factor = 70;
+
+	static const char *usage_str = "usage: %s "
+		"[-h] "
+		"[-p <pool_size>] "
+		"[-s <segment_size>] "
+		"[-v <val_max_factor>] "
+		"[-t <timeout_seconds>] "
+		"[-m <timeout_minutes>] "
+		"[-o <timeout_hours>] "
+		"[-d <dir>]\n ";
+
+	const char *optstr = "hp:s:v:t:m:o:d:";
+	int opt;
+	while ((opt = getopt(argc, argv, optstr)) != -1) {
+		char *endptr;
+		switch (opt) {
+		case 'h':
+			printf(usage_str, argv[0]);
+			exit(0);
+		case 'p':
+			p.pool_size = (size_t)(strtoumax(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid pool size provided");
+			break;
+		case 's':
+			p.segment_size =
+			    (size_t)(strtoumax(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid segment size provided");
+			break;
+		case 'v':
+			val_max_factor =
+			    (size_t)(strtoumax(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid val max factor ovided");
+			break;
+		case 't':
+			p.seconds = (int)(strtoul(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid seconds provided");
+			break;
+		case 'm':
+			p.minutes = (int)(strtoul(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid minutes provided");
+			break;
+		case 'o':
+			p.hours = (int)(strtoul(optarg, &endptr, 10));
+			if (strlen(endptr) != 0)
+				FATAL("invalid hours provided");
+			break;
+		case 'd':
+			strcpy(p.dir, optarg);
+			break;
+		}
+	}
+	p.val_max = val_max_factor * p.segment_size;
+
+	return p;
+}
+
+int
+main(int argc, char **argv)
+{
+	test_params p = parse_args(argc, argv);
+
+	/* init pool */
+	VMEMcache *vc = vmemcache_new(p.dir, p.pool_size, p.segment_size, 1);
+	if (!vc)
+		FATAL("vmemcache_new: %s", p.dir);
+
+	on_evict_info info = { 0, 0, false };
+	vmemcache_callback_on_evict(vc, on_evict, &info);
+
+	/* print csv header */
+	printf("keynum,ratio\n");
+
+	float prev_ratio;
+	float ratio = 0.0f;
+	bool print_ratio = false;
+
+	long seed = time(NULL);
+	srand((unsigned)seed);
+
+	char val;
+	size_t val_size;
+	char key[MAX_KEYSIZE];
+	size_t keynum = 0;
+
+	time_t start;
+	time(&start);
+
+	while (timeout(start, p.seconds, p.minutes, p.hours)) {
+		/* create key */
+		sprintf(key, "%zu", keynum);
+
+		/* generate val */
+		val_size = get_rand_size(p.val_max, p.segment_size);
+		info.allocated_size += val_size;
+
+		/* put */
+		int ret = vmemcache_put(vc, key, MAX_KEYSIZE, &val, val_size);
+		if (ret != 0) {
+			fprintf(stderr, "vmemcache_put ret: %d\n", ret);
+			goto err;
+		}
+
+		/*
+		 * Do not print the csv line if current ratio value is the same
+		 * (taking precision into account) as the previous one. The
+		 * intent is to avoid unnecessary bloating of the csv output.
+		 */
+		ratio = (float)info.allocated_size / (float)p.pool_size;
+		print_ratio = keynum == 0 || roundf(ratio * 100)
+			!= roundf(prev_ratio * 100);
```

Comment:
Comparing floats is... tricky. Maybe it works fine in this case because they are rounded values, maybe not.
lroundf?